### PR TITLE
feat: Adding web 3 login

### PR DESCRIPTION
**Pull request by Dudleyneedham**
## fixes https://github.com/KILTprotocol/ticket/issues/2945

Removing the next js and updating to the web 3 login

## How to test:

Go through the CLI and check if working

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)
    * Either PR or Ticket to update [the Docs](https://github.com/KILTprotocol/docs)
    * Link the PR/Ticket here

### CODE REVIEW

File: menus/setup-identity.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export default async function ({ returnAssets = false } = {}): Promise<any> {
   await status('connecting to network...')
   await connect(network)
   const account = await loadAccount(mnemonic)
-  const keypairs = await getKeypairs(account, mnemonic)
+  const keypairs = await getKeypairs(mnemonic)
```
Comment (kilted-andres) [inline menus/setup-identity.ts:37]: The calls of this function need to be updated to remove "account" from the parameters. 

Files that I found that need update:

- `menus/setup-identity.ts`
- `menus/setup-claimer.ts`
- `menus/create-ctype.ts`

File: menus/utils/utils.ts [hunk 2]
```diff
@@ -68,42 +58,48 @@ export async function isCtypeOnChain(ctype: ICType): Promise<boolean> {
...
+  const api = ConfigService.get('api')
+
+  let balance = parseInt(
+    (await api.query.system.account(account.address)).data.free.toString()
+  )
+  if (balance < 3) {
+    const message = testnet
+      ? chalk.red(
+          `get testnet tokens to continue => https://faucet.peregrine.kilt.io/?${account.address}`
+        )
```
Comment (kilted-andres) [inline menus/utils/utils.ts:71]: Maybe put orange here? 

Red is more after a problem occurs, like on line 72.

**Review by kilted-andres — CHANGES_REQUESTED** [review]
Good new features.

It does not really compile, I had to change some stuff to compile it. I will revert my commits now. 

For the set-up of web3-login, it would be better not to ask for the `origin`, but only for the frontend port. 

Compatibility with the new yarn should be also guaranteed.

**Comment by kilted-andres** [comment]
The author is still Paul. I would say Dudley could have that title now.

**Comment by kilted-andres** [comment]
For the set up of the `web3login` the creation of a `well-know-did-config` is missing. It would be nice to trigger it from the CLI as well. 

The command for that inside `web3-login-demo` is: `yarn well-known-config`, wich executes `./scripts/genesisWellKnownDidConfig.ts`

File: menus/utils/utils.ts (hunk 2 above)
Comment (Dudleyneedham) [reply menus/utils/utils.ts:71]:
```suggestion
    const message = testnet
      ? chalk.orange(
          `get testnet tokens to continue => https://faucet.peregrine.kilt.io/?${account.address}`
        )
```

**Comment by Dudleyneedham** [comment]
> For the set up of the `web3login` the creation of a `well-know-did-config` is missing. It would be nice to trigger it from the CLI as well.
> 
> The command for that inside `web3-login-demo` is: `yarn well-known-config`, wich executes `./scripts/genesisWellKnownDidConfig.ts`

Yes, this was before the unification. I would do it again now with the different keys being assigned corresponding with the new unified key generations across the various applications.

**Comment by Dudleyneedham** [comment]
> For the set up of the `web3login` the creation of a `well-know-did-config` is missing. It would be nice to trigger it from the CLI as well.
> 
> The command for that inside `web3-login-demo` is: `yarn well-known-config`, wich executes `./scripts/genesisWellKnownDidConfig.ts`

We may not want to add the well-known did configuration since it should be released with the upcoming extension library when it does get released. It should be changed throughout.

File: menus/setup-identity.ts (hunk 1 above)
Comment (Dudleyneedham) [reply menus/setup-identity.ts:39]: Done thank you :)

File: .yarnrc.yml [hunk 3]
```diff
@@ -1 +1,3 @@
 nodeLinker: node-modules
+
+yarnPath: .yarn/releases/yarn-3.6.4.cjs
```
Comment (kilted-andres) [inline .yarnrc.yml:3]: `.yarn` is part of the list inside `.gitignore`, so the files inside of `.yarn` are not part of the commited repo. 

While trying to run `yarn install` or any yarn command I get this error: 
```
node:internal/modules/cjs/loader:1051
  throw err;
  ^

Error: Cannot find module '/Users/andresrodriguez/Visual_Studio_Code/kilt-distillery-cli/.yarn/releases/yarn-3.6.4.cjs'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1048:15)
    at Module._load (node:internal/modules/cjs/loader:901:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

This is very inconvenient.

**Review by kilted-andres — CHANGES_REQUESTED** [review]
Cannot run `yarn install`

File: .yarnrc.yml (hunk 3 above)
Comment (Dudleyneedham) [reply .yarnrc.yml:3]: This is a problem with your Yarn version

File: menus/utils/initialiseProject.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+export async function initialiseProject(dappName: string) {
+  await status('initializing project...')
+  process.chdir(dappName)
+
+  await status('installing dependencies...')
+  await promisifyExec('yarn install')
+  await status('building...')
+  await promisifyExec('yarn run build')
+  await status('Creating a well known did config')
+  await promisifyExec('yarn run did-configuration')
```
Comment (kilted-andres) [inline menus/utils/initialiseProject.ts:16]: Very nice!

File: menus/_prompts.ts [hunk 5]
```diff
@@ -88,48 +101,54 @@ export async function useExisting() {
...
+            name: 'Spirit mainnet',
+            value: 'wss://spiritnet.api.onfinality.io/public-ws',
+          },
+        ],
+      },
     })
   ).network
 }
 
 export async function getOrigin() {
```
Comment (kilted-andres) [inline menus/_prompts.ts:123]: This should not be asked for the creation of a web3-login project. 

The origin is made from the Frontend port, which is ask to be typed right after this prompt.

**Review by kilted-andres — CHANGES_REQUESTED** [review]
Great improvements! I like this PR. And it is compiling in my computer now. :)

My only request is to skip the "origin" prompt. The project uses the Frontend port to craft ist origin: “http://localhost/:FRONTEND_PORT”. 

On the well-known-did-config inside the product-project the right url for the origin is being used.

**Review by kilted-andres — APPROVED** [review]
Everything is great! 

Nice, nice, nice.

**Pull request merged**